datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -170,7 +171,17 @@ def __init__(self, instance, init_config, remapper=None, logger=None):
         auth = None
         if config['password']:
             if config['username']:
-                auth = (config['username'], config['password'])
+                if config['auth_type']:
+                    auth_type = config.get('auth_type', 'basic').lower()
+                    if auth_type not in ('basic', 'digest'):
```

Comment:
Let's check for each then log in an `else` branch. Also, make the first be `if auth_type == 'basic':`